docs/src/pages/guides/minimizing-bundle-size/minimizing-bundle-size.md

```diff
@@ -76,6 +76,79 @@ However, you need to apply the following steps correctly.
 
 Pick one of the following plugins:
 
+- [babel-plugin-transform-imports](https://www.npmjs.com/package/babel-plugin-transform-imports) with the following configuration:
+
+  `yarn add -D babel-plugin-transform-imports`
+
+  Create a `.babelrc` file in the root directory of your project:
+  
+  ```js
+  {
+    "plugins": [
+      [
+        "transform-imports",
+        {
+          "@material-ui/core": {
+            // for bundlers not supporting ES modules use:
+            // transform: '@material-ui/core/${member}',
+            "transform": "@material-ui/core/esm/${member}",
+            "preventFullImport": true
+          },
+          "@material-ui/icons": {
+            // for bundlers not supporting ES modules use:
+            // transform: '@material-ui/core/${member}',
+            "transform": "@material-ui/icons/esm/${member}",
+            "preventFullImport": true
+          }
+        }
+      ]
+    ]
+  }
+  ```
+  
+  If you are using Create React App, you will need to use a couple projects that let you use a `.babelrc` file, without ejecting. 
+  
+  `yarn add -D react-app-rewired customize-cra`
+  
+  Create a `config-overrides.js` file in the root directory:
+
+  ```js
+  /* config-overrides.js */
+  const { useBabelRc, override } = require('customize-cra')
+
+  module.exports = override(
+    useBabelRc()
+  );  
+  ```
+  
+  Modify your `package.json` start command:
+  
+```diff
+  "scripts": {
+-  "start": "react-scripts start"
++  "start": "react-app-rewired start"
+  }
+```
+  
+  Note: You may run into errors like these:
+
+  ```
+    Module not found: Can't resolve '@material-ui/core/makeStyles' in '/your/project'
+    Module not found: Can't resolve '@material-ui/core/createStyles' in '/your/project'
```

Comment:
This sounds like a limitation of the plugin. The proposed fix is not equivalent. Did you consider using the other plugin?

Reply:
I did not try the other plugin.

I don't understand how the proposed fix is different.

https://material-ui.com/styles/api/#createstyles-styles-styles

Uses `import {makeStyles, createStyles} from '@material-ui/styles'`

Reply:
Weird, I think I might be seeing it now.

theme is suddenly an empty object for me in my project... =(

import {makeStyles, createStyles} from "@material-ui/styles";
const useStyles = makeStyles((theme: Theme) =>
	createStyles()
);

Reply:
`@material-ui/styles` has no default theme, `@material-ui/core/styles` does

Reply:
Oh wild, I disabled the plugin entirely and moving create/makeStyles from /core /styles causes the little code snippet above to fail. Theme object is null. This is contrary to the documentation I quoted above. @oliviertassinari any links for more information on this? My app is a standard CRA.

Reply:
Ah... https://stackoverflow.com/questions/53925829/react-material-ui-theme-undefined-in-createstyles-with-typescript

Wow, so confusing!

Reply:
Ok, I've tried both plugins now and neither one works with the documented configuration. They both puke on finding create/makeStyles.

Reply:
It's documented here https://material-ui.com/customization/default-theme/#material-ui-core-styles-vs-material-ui-styles

Reply:
@merceyz That is super helpful, thanks! I just wish the same exact names hadn't been used. It is far too easy to mess up an import.

Now, any suggestions on making these plugins work? I'll happily rewrite the documentation and update the PR, once we get this figured out. The startup speed increase is quite noticeable, so I think the effort is worth it.

FYI, I'm not noticing as much speed increase for just the icons with `babel-plugin-import` as I am with the other one.

Reply:
Ok, figured it out. Just as you said, use the right import. core/styles... That solves it.

Reply:
> Now, any suggestions on making these plugins work? I'll happily rewrite the documentation and update the PR, once we get this figured out. The startup speed increase is quite noticeable, so I think the effort is worth it.

Could take the same aproach as lodash and make our own plugin

Reply:
Meh, I feel that this documentation update is enough. More code is more to maintain.